File: app/udptransport/src/main/java/org/exthmui/share/udptransport/packets/CommandPacket.java
package org.exthmui.share.udptransport.packets;

import androidx.annotation.NonNull;

import java.net.DatagramPacket;

public final class CommandPacket extends AbstractCommandPacket<CommandPacket> {
    public CommandPacket(DatagramPacket packet) {
        super(packet);
    }

    public CommandPacket() {
        super();
    }

    @Override
    public void legalCheck() {
        super.legalCheck();
    }

    @NonNull
    @Override
    public CommandPacket clone() {
        return new CommandPacket(this.toDatagramPacket());
    }

    public static CommandPacket of(DatagramPacket packet) {
        return new CommandPacket(packet);
    }


}

File: app/udptransport/src/main/java/org/exthmui/share/udptransport/packets/CommandPacket.kt
package org.exthmui.share.udptransport.packets

import java.net.DatagramPacket

class CommandPacket : AbstractCommandPacket<CommandPacket?> {
    constructor(packet: DatagramPacket?) : super(packet!!)
    constructor() : super()

    override fun legalCheck() {
        super.legalCheck()
    }

    public override fun clone(): CommandPacket {
        return CommandPacket(toDatagramPacket())
    }

    companion object {
        fun of(packet: DatagramPacket?): CommandPacket {
            return CommandPacket(packet)
        }
    }
}
